parity/db/rocksdb/helpers.rs

```diff
@@ -27,10 +27,38 @@ pub fn compaction_profile(profile: &DatabaseCompactionProfile, db_path: &Path) -
 	}
 }
 
+pub fn memory_per_column(total: Option<usize>) -> HashMap<Option<u32>, usize> {
+	let mut memory_per_column = HashMap::new();
+	if let Some(budget) = total {
+		// spend 90% of the memory budget on the state column, but at least 256 MiB
+		memory_per_column.insert(ethcore_db::COL_STATE, std::cmp::max(budget * 9 / 10, 256));
+		let num_columns = ethcore_db::NUM_COLUMNS.expect("NUM_COLUMNS is Some; qed");
+		// spread the remaining 10% evenly across columns
+		let rest_budget = budget / 10 / (num_columns as usize - 1);
```

Comment:
why `num_columns as usize - 1` here?

Reply:
-1 for the state column

Reply:
ah, it should probably be -1 for the light db as well